### PR TITLE
uniform makefile experience

### PR DESCRIPTION
**Pull request by wangxinyi7**
Make uniform dev experience of building image. 
Ticket: https://hashicorp.atlassian.net/browse/NET-4834

### CODE REVIEW

**Comment by analogue** [comment]
@wangxinyi7 Please add reviewers that have vested interest and context.  In this case, @wilkermichael would be ideal since he created NET-4834 and took the initiative to track this work.

**Comment by wangxinyi7** [comment]
> @wangxinyi7 Please add reviewers that have vested interest and context. In this case, @wilkermichael would be ideal since he created NET-4834 and took the initiative to track this work.

@analogue I think he is in vacation so didn't tag him. Thanks @curtbushko for stepping in to review this PR.

**Comment by analogue** [comment]
> > @wangxinyi7 Please add reviewers that have vested interest and context. In this case, @wilkermichael would be ideal since he created NET-4834 and took the initiative to track this work.
> 
> @analogue I think he is in vacation so didn't tag him. Thanks @curtbushko for stepping in to review this PR.

Even more reason to tag @wilkermichael since you're leaving breadcrumbs for him to follow on his return should he wish to get up to speed.

**Review by curtbushko — CHANGES_REQUESTED** [review]
Looks pretty good. 

Feel free to say no to the DOCKER_TARGET request =)

File: Makefile [hunk 1]
```diff
@@ -23,31 +26,25 @@ GOLDFLAGS=-X github.com/hashicorp/consul-dataplane/pkg/version.GitCommit=$(GIT_C
...
-BUILD_ARGS = BIN_NAME=$(BIN_NAME) PRODUCT_VERSION=$(VERSION) PRODUCT_REVISION=$(REVISION)
-TAG        = $(PRODUCT_NAME)/$(TARGET):$(VERSION)
-BA_FLAGS   = $(addprefix --build-arg=,$(BUILD_ARGS))
-FLAGS      = --target $(TARGET) --platform $(PLATFORM) --tag $(TAG) $(BA_FLAGS)
-
 # Set OS to linux for all docker/* targets.
-docker/%: OS = linux
+docker/%: OS = linux## build docker images depending on the target
 
 # DOCKER_TARGET is a macro that generates the build and run make targets
```
[the remark this answers is not in the document]
Comment (wangxinyi7) [reply Makefile]: Yup, removed the docker_target macro

File: Makefile [hunk 2]
```diff
@@ -71,18 +68,54 @@ $(eval $(call DOCKER_TARGET,release-default,bin))
 $(eval $(call DOCKER_TARGET,release-ubi,bin))
 
 .PHONY: docker
-docker: docker/release-default
+docker: docker/release-default ## build release-default target
```
[the remark this answers is not in the document]
Comment (wangxinyi7) [reply Makefile]: Sorry I think I missed some context here and didn't find the example in the consul-k8s so I didn't update this, maybe could do it in another PR? Or feel free to commit in the PR if it's straightforward to you.

**Comment by curtbushko** [comment]
@wangxinyi7 It looks like the integration tests are failing.

**Comment by curtbushko** [comment]
@wangxinyi7 This one too should also be backported all the way to the older release branches

**Pull request merged**